### PR TITLE
arrays-multidimensional_arrays-hard-q4

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/multidimensional_arrays/hard/q4/AlterTable.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package arrays.multidimensional_arrays.hard.q4;
+
+/*
+Make a program that defines a 2D array and goes through each element in it by traversing each row's elements one at a
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q4/AlterTable.java:4]: Change question text to:
"Make a program that defines a multidimensional array and iterates over each row of the array, printing out every other element starting from the first element of the first row. For example, if you have the following 2D array {{1,2,3}, {4, 5 ,6}, {7,8,9} }, the program will print 1, 3, 5, 7, 9."

File: src/arrays/multidimensional_arrays/hard/q4/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package arrays.multidimensional_arrays.hard.q4;
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public static int[][] arr = {{4, 5, 6}, {66, 54, 21}, {609, 998, 543}, {1111, 2342, 4583}};
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q4/MainTest.java:10]:
```suggestion
```

File: src/arrays/multidimensional_arrays/hard/q4/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral(answerFor(arr))
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q4/MainTest.java:14]: The answer here should be hardcoded in as it will never change.

File: src/arrays/multidimensional_arrays/hard/q4/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        };
+    }
+
+    public void runMain() {
+        AlterTable.main(new String[0]);
+    }
+
+    public static String answerFor(int[][] nums) {
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q4/MainTest.java:22]: This helper method is not required for the question.

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/arrays/multidimensional_arrays/hard/q4/AlterTable.java (hunk 1 above)
Comment (SamamthaH) [reply src/arrays/multidimensional_arrays/hard/q4/AlterTable.java:4]: The question text was not changed

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Review by SamamthaH — APPROVED** [review]
